macro/run_sim_aligen.C

```diff
@@ -0,0 +1,169 @@
+#if !defined(__CLING__) || defined(__ROOTCLING__)
```

Comment:
This file comes with a very general name but only supports two modes depending on `USE_ALIGENHIJING`. Since the simulation has anyhow moved away from standalone macros how about dropping the file and include the relevant code to the `GeneratorFactory`?

The cmake setup checks for availability of AliRoot and sets `AliRoot_FOUND` if present. This can be used for a conditional compilation of the generator instantiation in the factory.

Reply:
hi @matthiasrichter , 
`run_sim_aligen.C` is indeed supposed to be generic, being an example macro to show how the added functionality enables one to use any generic generator that inherits from the `TGenerator` ROOT class (as it is the case for all the generators implemented in AliRoot).

The code can be anyway added in `GeneratorFactory` with the same example philosophy.